### ISSUE TITLE
Install error: unrecognized command-line option '-rpath'

### ISSUE THREAD

**Issue by bensworth** [firedrakeproject/firedrake#2000]
I am trying to install Firedrake using the python install script. I make pretty good progress, then fail on SuiteSparse in Petsc. Overview, it complains about `unrecognized command-line option '-rpath'`. I'm not sure if I'm having an issue with Firedrake, Petsc, or SuiteSparse. I think I found the offending command:

`/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicc -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -dynamiclib -single_module -undefined dynamic_lookup -multiply_defined suppress -I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -dynamiclib -compatibility_version 5 -current_version 5.8.1 -Wl,-install_name -Wl,@rpath/libsuitesparseconfig.5.dylib -shared -undefined dynamic_lookup SuiteSparse_config.o -o /Users/southworth/Software/firedrake/src/petsc/default/lib/libsuitesparseconfig.5.8.1.dylib -lm -rpath /Users/southworth/Software/firedrake/src/petsc/default/lib`

I'm not an expert on installations,  but some googling suggests that that spaces around `-rpath` are not correct/not what the command should look like. Everywhere else it has commas surrounding, e.g., `-lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar...` But, I have no idea how to fix this or where it is happening (Firedrake, Petsc, or Suitesparse). Any help is greatly appreciated. Below are more complete details in case useful.

In terminal, I get 
```
Traceback (most recent call last):
  File "/Users/southworth/Software/firedrake-install", line 1568, in <module>
    install("petsc/")
  File "/Users/southworth/Software/firedrake-install", line 835, in install
    build_and_install_petsc()
  File "/Users/southworth/Software/firedrake-install", line 969, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + list(petsc_options))
  File "/Users/southworth/Software/firedrake-install", line 616, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
  File "/usr/local/Cellar/python@3.9/3.9.1_8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/local/Cellar/python@3.9/3.9.1_8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/southworth/Software/firedrake/bin/python', './configure', 'PETSC_DIR=/Users/southworth/Software/firedrake/src/petsc', 'PETSC_ARCH=default', '--download-chaco', '--download-hdf5', '--with-fortran-bindings=0', '--with-shared-libraries=1', '--download-mpich-configure-arguments=--disable-opencl', '--with-debugging=0', '--download-pastix', '--download-hwloc', '--download-hypre', '--download-mumps', '--download-mpich', '--with-c2html=0', '--download-ptscotch', '--with-cxx-dialect=C++11', '--download-ml', '--download-eigen=/Users/southworth/Software/firedrake/src/eigen-3.3.3.tgz ', '--with-blaslapack-dir=/usr/local/opt/openblas', '--download-suitesparse', '--download-netcdf', '--download-metis', '--LDFLAGS=-Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib', '--download-hwloc-configure-arguments=--disable-opencl', '--CFLAGS=-I/usr/local/opt/openblas/include -Wno-implicit-function-declaration', '--download-scalapack', '--download-pnetcdf', '--with-zlib', '--download-superlu_dist']' returned non-zero exit status 1.

Install log saved in /Users/southworth/Software/firedrake-install.log
```
The firedrake-install.log looks like:
```
*******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
Error running make on SuiteSparse: Could not execute "['cd /Users/southworth/Software/firedrake/src/petsc/default/externalpackages/git.suitesparse/SuiteSparse_config && /usr/local/bin/gmake CC="/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicc" CXX="/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicxx" MAKE="/usr/local/bin/gmake" RANLIB="/usr/bin/ranlib -c" ARCHIVE="/usr/bin/ar cr" RM="/bin/rm -f" MV="/bin/mv" CP="/bin/cp" LDFLAGS="-Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -dynamiclib -single_module -undefined dynamic_lookup -multiply_defined suppress -I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -L\\$(INSTALL_LIB)" INSTALL_LIB=/Users/southworth/Software/firedrake/src/petsc/default/lib INSTALL_INCLUDE=/Users/southworth/Software/firedrake/src/petsc/default/include INSTALL_DOC=/Users/southworth/Software/firedrake/src/petsc/default/share/doc/suitesparse BLAS="-Wl,-rpath,/usr/local/opt/openblas -L/usr/local/opt/openblas -lopenblas -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lstdc++ -ldl -lmpifort -lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -Wl,-rpath,/usr/local/opt/openblas/lib -lgfortran -lquadmath -lm" LAPACK="-Wl,-rpath,/usr/local/opt/openblas -L/usr/local/opt/openblas -lopenblas -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lstdc++ -ldl -lmpifort -lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -Wl,-rpath,/usr/local/opt/openblas/lib -lgfortran -lquadmath -lm" UMFPACK_CONFIG= MY_METIS_INC="/Users/southworth/Software/firedrake/src/petsc/default/include" MY_METIS_LIB="-Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lmetis" CF="-I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O" CHOLMOD_CONFIG="" CUDA=no CUDA_PATH= clean && /usr/local/bin/gmake CC="/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicc" CXX="/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicxx" MAKE="/usr/local/bin/gmake" RANLIB="/usr/bin/ranlib -c" ARCHIVE="/usr/bin/ar cr" RM="/bin/rm -f" MV="/bin/mv" CP="/bin/cp" LDFLAGS="-Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -dynamiclib -single_module -undefined dynamic_lookup -multiply_defined suppress -I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -L\\$(INSTALL_LIB)" INSTALL_LIB=/Users/southworth/Software/firedrake/src/petsc/default/lib INSTALL_INCLUDE=/Users/southworth/Software/firedrake/src/petsc/default/include INSTALL_DOC=/Users/southworth/Software/firedrake/src/petsc/default/share/doc/suitesparse BLAS="-Wl,-rpath,/usr/local/opt/openblas -L/usr/local/opt/openblas -lopenblas -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lstdc++ -ldl -lmpifort -lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -Wl,-rpath,/usr/local/opt/openblas/lib -lgfortran -lquadmath -lm" LAPACK="-Wl,-rpath,/usr/local/opt/openblas -L/usr/local/opt/openblas -lopenblas -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lstdc++ -ldl -lmpifort -lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -Wl,-rpath,/usr/local/opt/openblas/lib -lgfortran -lquadmath -lm" UMFPACK_CONFIG= MY_METIS_INC="/Users/southworth/Software/firedrake/src/petsc/default/include" MY_METIS_LIB="-Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lmetis" CF="-I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O" CHOLMOD_CONFIG="" CUDA=no CUDA_PATH= && /usr/local/bin/gmake CC="/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicc" CXX="/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicxx" MAKE="/usr/local/bin/gmake" RANLIB="/usr/bin/ranlib -c" ARCHIVE="/usr/bin/ar cr" RM="/bin/rm -f" MV="/bin/mv" CP="/bin/cp" LDFLAGS="-Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -dynamiclib -single_module -undefined dynamic_lookup -multiply_defined suppress -I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -L\\$(INSTALL_LIB)" INSTALL_LIB=/Users/southworth/Software/firedrake/src/petsc/default/lib INSTALL_INCLUDE=/Users/southworth/Software/firedrake/src/petsc/default/include INSTALL_DOC=/Users/southworth/Software/firedrake/src/petsc/default/share/doc/suitesparse BLAS="-Wl,-rpath,/usr/local/opt/openblas -L/usr/local/opt/openblas -lopenblas -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lstdc++ -ldl -lmpifort -lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -Wl,-rpath,/usr/local/opt/openblas/lib -lgfortran -lquadmath -lm" LAPACK="-Wl,-rpath,/usr/local/opt/openblas -L/usr/local/opt/openblas -lopenblas -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lstdc++ -ldl -lmpifort -lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -Wl,-rpath,/usr/local/opt/openblas/lib -lgfortran -lquadmath -lm" UMFPACK_CONFIG= MY_METIS_INC="/Users/southworth/Software/firedrake/src/petsc/default/include" MY_METIS_LIB="-Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lmetis" CF="-I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O" CHOLMOD_CONFIG="" CUDA=no CUDA_PATH= install && /usr/local/bin/gmake CC="/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicc" CXX="/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicxx" MAKE="/usr/local/bin/gmake" RANLIB="/usr/bin/ranlib -c" ARCHIVE="/usr/bin/ar cr" RM="/bin/rm -f" MV="/bin/mv" CP="/bin/cp" LDFLAGS="-Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -dynamiclib -single_module -undefined dynamic_lookup -multiply_defined suppress -I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -L\\$(INSTALL_LIB)" INSTALL_LIB=/Users/southworth/Software/firedrake/src/petsc/default/lib INSTALL_INCLUDE=/Users/southworth/Software/firedrake/src/petsc/default/include INSTALL_DOC=/Users/southworth/Software/firedrake/src/petsc/default/share/doc/suitesparse BLAS="-Wl,-rpath,/usr/local/opt/openblas -L/usr/local/opt/openblas -lopenblas -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lstdc++ -ldl -lmpifort -lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -Wl,-rpath,/usr/local/opt/openblas/lib -lgfortran -lquadmath -lm" LAPACK="-Wl,-rpath,/usr/local/opt/openblas -L/usr/local/opt/openblas -lopenblas -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lstdc++ -ldl -lmpifort -lmpi -lpmpi -lgfortran -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10/gcc/x86_64-apple-darwin19/10.2.0 -Wl,-rpath,/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -L/usr/local/Cellar/gcc/10.2.0_4/lib/gcc/10 -Wl,-rpath,/usr/local/opt/openblas/lib -lgfortran -lquadmath -lm" UMFPACK_CONFIG= MY_METIS_INC="/Users/southworth/Software/firedrake/src/petsc/default/include" MY_METIS_LIB="-Wl,-rpath,/Users/southworth/Software/firedrake/src/petsc/default/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -lmetis" CF="-I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O" CHOLMOD_CONFIG="" CUDA=no CUDA_PATH= clean']":
( cd xerbla ; /usr/local/bin/gmake clean )
gmake[1]: Entering directory '/Users/southworth/Software/firedrake/src/petsc/default/externalpackages/git.suitesparse/SuiteSparse_config/xerbla'
/bin/rm -f -r *.o *.obj *.ln *.bb *.bbg *.da *.tcov *.gcov gmon.out *.bak *.d *.gcda *.gcno *.aux *.bbl *.blg *.log *.toc *.dvi *.lof *.lot
gmake[1]: Leaving directory '/Users/southworth/Software/firedrake/src/petsc/default/externalpackages/git.suitesparse/SuiteSparse_config/xerbla'
/bin/rm -f -r *.o *.obj *.ln *.bb *.bbg *.da *.tcov *.gcov gmon.out *.bak *.d *.gcda *.gcno *.aux *.bbl *.blg *.log *.toc *.dvi *.lof *.lot
/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicc -I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O -c SuiteSparse_config.c
/usr/bin/ar cr libsuitesparseconfig.a SuiteSparse_config.o
/usr/bin/ranlib -c libsuitesparseconfig.a
/usr/local/bin/gmake install INSTALL=/Users/southworth/Software/firedrake/src/petsc/default/externalpackages/git.suitesparse
gmake[1]: Entering directory '/Users/southworth/Software/firedrake/src/petsc/default/externalpackages/git.suitesparse/SuiteSparse_config'
/Users/southworth/Software/firedrake/src/petsc/default/bin/mpicc -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -dynamiclib -single_module -undefined dynamic_lookup -multiply_defined suppress -I/usr/local/opt/openblas/include -Wno-implicit-function-declaration -g -O -Wl,-rpath,/usr/local/opt/openblas/lib -L/usr/local/opt/openblas/lib -L/Users/southworth/Software/firedrake/src/petsc/default/lib -dynamiclib -compatibility_version 5 -current_version 5.8.1 -Wl,-install_name -Wl,@rpath/libsuitesparseconfig.5.dylib -shared -undefined dynamic_lookup SuiteSparse_config.o -o /Users/southworth/Software/firedrake/src/petsc/default/lib/libsuitesparseconfig.5.8.1.dylib -lm -rpath /Users/southworth/Software/firedrake/src/petsc/default/lib
gmake[1]: Leaving directory '/Users/southworth/Software/firedrake/src/petsc/default/externalpackages/git.suitesparse/SuiteSparse_config'gcc: error: unrecognized command-line option '-rpath'
gmake[1]: *** [Makefile:53: /Users/southworth/Software/firedrake/src/petsc/default/lib/libsuitesparseconfig.5.8.1.dylib] Error 1
gmake: *** [Makefile:23: library] Error 2
*******************************************************************************


2021-03-08 17:03:59,048 DEBUG  Popping path '/Users/southworth/Software/firedrake/src/petsc'
2021-03-08 17:03:59,048 DEBUG  New path '/Users/southworth/Software/firedrake/src'
2021-03-08 17:03:59,082 INFO   

Install log saved in /Users/southworth/Software/firedrake-install.log
```

**Comment by wence-** [comment]
Hi Ben, interesting. I tried this on my mac and the link runs fine.

It looks like your system is using some version of GCC, rather than Clang, for the compiler inside MPICC.
```
gcc: error: unrecognized command-line option '-rpath'
```

In `firedrake/src/petsc` can you say:

```
$ PETSC_DIR=`pwd` PETSC_ARCH="default" make info
```

That might give us a clue as to how you ended up picking up GCC, rather than clang.

**Comment by bensworth** [comment]
Hi @wence-, Petsc team was able to identify something inside of Petsc that probably caused this (see https://gitlab.com/petsc/petsc/-/issues/855#note_524836810). There is a fix on a Petsc branch. Is there an easy way to have firedrake install a branch of Petsc from the firedrake-install script?

Re compiler, I installed a brew gcc for compiling because the clang one was not playing nicely with mpi compilers I installed (other non firedrake code).
